src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetExceptionForHRTests.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public class GetExceptionForHRTests
+    {
+        public static readonly object[][] TestData =
+        {
+            new object[] { unchecked((int)0x80020006) },
+            new object[] { unchecked((int)0x80020101) }
+        };
+
+        [Theory]
+        [MemberData(nameof(TestData))]
```

Comment:
You can just use [InlineData(...)] for these instead.

Reply:
@jiangzeng If I recall this was one of the test you mentioned was 'randomly' failing. I wasn't able to reproduce this in a simple EXE project, but debugging through the `Marshal.GetExceptionForHR()` seemed to indicate there was some thread specific state that was queried. @luqunl Do you have any insight into how we could test this function in a more reliable way?

Reply:
@stephentoub Fixed with inlinedata, see in next commit